ocaml/tests/test_ha_vm_failover.ml

```diff
@@ -495,11 +509,62 @@ module AssertNewVMPreservesHAPlan = Generic.Make(Generic.EncapsulateState(struct
                                                    ]
                                                  end))
 
+module ComputeMaxFailures = Generic.Make(Generic.EncapsulateState(struct
+                                            module Io = struct
+                                              open Xapi_ha_vm_failover
+
+                                              type input_t = pool
+                                              type output_t = int
+
+                                              let string_of_input_t = string_of_pool
+                                              let string_of_output_t = string_of_int
+                                            end
+
+                                            module State = Test_state.XapiDb
+
+                                            let load_input __context = setup ~__context
+
+                                            let extract_output __context pool =
+                                              let max_hosts = Xapi_ha_vm_failover.compute_max_host_failures_to_tolerate ~__context () in
+                                              (* Struct requires input_t but not used here *)
+                                              pool |> ignore;
+                                              Int64.to_int max_hosts
+         
+                                            let tests = [
+                                              (* Three host pool with no VMs. *)
+                                              (
+                                                {
+                                                  master = {memory_total = gib 256L; name_label = "master"; vms = []};
+                                                  slaves = [
+                                                    {memory_total = gib 256L; name_label = "slave1"; vms = []};
+                                                    {memory_total = gib 256L; name_label = "slave2"; vms = []}
+                                                  ];
+                                                  ha_host_failures_to_tolerate = 3L;
+                                                  cluster = true;
+                                                },
+                                                1
+                                              );
+                                              (* Two hosts pool with no VMs  *)
+                                              (
+                                                {
+                                                  master = {memory_total = gib 256L; name_label = "master"; vms = []};
+                                                  slaves = [
+                                                    {memory_total = gib 256L; name_label = "slave1"; vms = []}
+                                                  ];
+                                                  ha_host_failures_to_tolerate = 2L;
```

Comment:
If your pool only has 2 hosts how can you loose them all? I think you need some actual VMs for the tests to make sense, indeed if you have no VMs you  might as well turn the whole pool off, but thats not a realistic scenario.